seabird.proto

```diff
@@ -4,168 +4,163 @@ option go_package = ".;pb";
 
 package seabird;
 
-// Identity is included in every message that requires auth. One AuthMethod at a
-// time is supported. For now, the only supported method is a token. These need
-// to be created manually and distributed to any clients.
-message Identity {
-  oneof AuthMethod {
-    string token = 1;
-  }
-}
-
-service Seabird {
-  rpc StreamEvents(StreamEventsRequest) returns (stream Event);
+import "common.proto";
 
-  // Chat actions
-  rpc SendMessage(SendMessageRequest) returns (SendMessageResponse);
-  rpc JoinChannel(JoinChannelRequest) returns (JoinChannelResponse);
-  rpc LeaveChannel(LeaveChannelRequest) returns (LeaveChannelResponse);
+// Note that the IDs exposed by the Service, User, and Channel are compound IDs
+// which seabird-core will parse and route to the correct location.
 
-  // Chat connection introspection
-  rpc ListChannels(ListChannelsRequest) returns (ListChannelsResponse);
-  rpc GetChannelInfo(ChannelInfoRequest) returns (ChannelInfoResponse);
-  rpc SetChannelTopic(SetChannelTopicRequest) returns (SetChannelTopicResponse);
-
-  // Seabird introspection
-  rpc ListStreams(ListStreamsRequest) returns (ListStreamsResponse);
-  rpc GetStreamInfo(StreamInfoRequest) returns (StreamInfoResponse);
-  rpc GetCoreInfo(CoreInfoRequest) returns (CoreInfoResponse);
+message Event {
+  oneof inner {
+    common.MessageEvent message = 1;
+    common.PrivateMessageEvent private_message = 2;
+    common.MentionEvent mention = 3;
+    common.CommandEvent command = 4;
+  }
 }
 
+// Request and response objects
+
 message StreamEventsRequest {
-  Identity identity = 1;
+  common.Identity identity = 1;
 
+  // A registry of commands this plugin responds to. A plugin MUST register all
+  // commands it responds to in order to receive those events. The
```

Comment:
`...order to receive those events as commands instead of MessageEvents`, right?

Reply:
No, if a message is parsed as a command, it will not be another message type.

Reply:
I'd add that here as well then.

Reply:
Please add a note similar to what you added in the command message comment